backend/gateway/gateway.go

```diff
@@ -340,5 +341,33 @@ func RunWithConfig(f *Flags, cfg *gatewayv1.Config, cf *ComponentFactory, assets
 		ReadTimeout:  timeout,
 		WriteTimeout: timeout,
 	}
-	logger.Fatal("error bringing up listener", zap.Error(srv.ListenAndServe()))
+
+	sc := make(chan os.Signal, 1)
+	signal.Notify(
+		sc,
+		os.Interrupt,
+		syscall.SIGTERM,
+		syscall.SIGQUIT,
+	)
+
+	go func() {
+		if err = srv.ListenAndServe(); err != http.ErrServerClosed {
+			// Only log an error if it's not due to shutdown or close
+			logger.Fatal("error bringing up listener", zap.Error(err))
+		}
+	}()
+
+	<-sc
+
+	signal.Stop(sc)
+
+	// Shutdown timeout should be max request timeout (with 1s buffer).
+	ctxShutDown, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	if err = srv.Shutdown(ctxShutDown); err != nil {
+		logger.Fatal("server shutdown failed", zap.Error(err))
+	}
+
+	logger.Debug("server shutdown gracefully")
```

Comment:
should we make this an info log instead? was thinking that'd it be useful to see when testing the change on staging. but i guess the fatal log above would be enough

Reply:
Yeah I wasn't sure if this would be too noisy. I suppose we implicitly log that the server started through the tcp bindings

Reply:
sounds good!